src/s2wasm.h

```diff
@@ -1193,6 +1198,42 @@ class S2WasmBuilder {
         }
       }
     }
+    if (!!startFunction) {
```

Comment:
`startFunction` is initialized with a `std::string` which makes it `"\0"` because `Name` passes that to `IString`'s ctor. So yes, it's a valid zero-length string :-)

I much prefer idiomatic `std::string`, despite its shortcomings! Commoning strings is fine in some cases, but I'm not sure these tools really need it especially for such short strings which could just be inline. We could also use `folly::string` which is more configurable and closer to `std::string`. Also well tested, but I'm biased on that ;-)